### DISCUSSION TITLE
Issues when train the Tacotron model on LJSpeech dataset.

### DISCUSSION THREAD

**Discussion by thomasvonl** [coqui-ai/TTS#587] [General Q&A]
I have chang the last two cell of  TTS_example.ipynb to 
`!python /content/TTS/TTS/bin/compute_statistics.py /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json /content/TTS/scale_stats.npy --data_path /content/LJSpeech-1.1/wavs/'
'!CUDA_VISIBLE_DEVICES="0" python /content/TTS/TTS/bin/train_tacotron.py --config_path /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json \ --coqpit.output_path ./Results \ --coqpit.datasets.0.path /content/LJSpeech-1.1/ \ --coqpit.audio.stats_path /content/TTS/scale_stats.npy \`

and  I have get the following error message
`Using CUDA: True
Number of GPUs: 1
Mixed precision mode is ON
fatal: not a git repository (or any of the parent directories): .git
Git Hash: 0000000
Experiment folder: DEFINE THIS/ljspeech-ddc-June-20-2021_02+22PM-0000000
fatal: not a git repository (or any of the parent directories): .git
Traceback (most recent call last):
File "/content/TTS/TTS/bin/train_tacotron.py", line 737, in
args, config, OUT_PATH, AUDIO_PATH, c_logger, tb_logger = init_training(sys.argv)
File "/content/TTS/TTS/utils/arguments.py", line 182, in init_training
config, OUT_PATH, AUDIO_PATH, c_logger, tb_logger = process_args(args)
File "/content/TTS/TTS/utils/arguments.py", line 168, in process_args
copy_model_files(config, experiment_path, new_fields)
File "/content/TTS/TTS/utils/io.py", line 42, in copy_model_files
copy_stats_path,
File "/usr/lib/python3.7/shutil.py", line 120, in copyfile
with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'scale_stats.npy'`

The file scale_stats.npy exists and the file path is correct. Is the -- coqpit.addio.stats_path argument set the state file path? And I track the error to the "utils/io.py" file and test the value of "config.audio.stats_path", I assume it should be "/content/TTS/scale_stats.npy " but it is "scale_stats.npy". And I run the ipynb file on the Colab, so it clone the latest repo.
@Sadam1195

**Comment by Sadam1195** [comment]
Have you tried running it with

```
'!CUDA_VISIBLE_DEVICES="0" python /content/TTS/TTS/bin/train_tacotron.py  \
--config_path /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json \ 
--coqpit.output_path ./Results \ 
--coqpit.datasets.0.path /content/LJSpeech-1.1/ \ 
```

try it without adding ` \ --coqpit.audio.stats_path /content/TTS/scale_stats.npy \` to the command

**Reply by thomasvonl** [reply]
sadly the error message is the same as before

**Reply by Sadam1195** [reply]
> sadly the error message is the same as before

please post your error log with modified command.

**Reply by thomasvonl** [reply]
The modified command is 
`# pull the trigger
!CUDA_VISIBLE_DEVICES="0" python /content/TTS/TTS/bin/train_tacotron.py  --config_path /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json \ --coqpit.output_path ./Results \ --coqpit.datasets.0.path ./content/LJSpeech-1.1/ 
`

and the error log is 
` > Using CUDA:  True
 > Number of GPUs:  1
   >  Mixed precision mode is ON
fatal: not a git repository (or any of the parent directories): .git
 > Git Hash: 0000000
 > Experiment folder: DEFINE THIS/ljspeech-ddc-June-23-2021_01+10PM-0000000
fatal: not a git repository (or any of the parent directories): .git
Traceback (most recent call last):
  File "/content/TTS/TTS/bin/train_tacotron.py", line 737, in <module>
    args, config, OUT_PATH, AUDIO_PATH, c_logger, tb_logger = init_training(sys.argv)
  File "/content/TTS/TTS/utils/arguments.py", line 182, in init_training
    config, OUT_PATH, AUDIO_PATH, c_logger, tb_logger = process_args(args)
  File "/content/TTS/TTS/utils/arguments.py", line 168, in process_args
    copy_model_files(config, experiment_path, new_fields)
  File "/content/TTS/TTS/utils/io.py", line 42, in copy_model_files
    copy_stats_path,
  File "/usr/lib/python3.7/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'scale_stats.npy'`

**Reply by Sadam1195** [reply]
> The modified command is
> `# pull the trigger !CUDA_VISIBLE_DEVICES="0" python /content/TTS/TTS/bin/train_tacotron.py --config_path /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json \ --coqpit.output_path ./Results \ --coqpit.datasets.0.path ./content/LJSpeech-1.1/ `
> 
> and the error log is
> ` > Using CUDA: True
> 
> > Number of GPUs:  1
> > Mixed precision mode is ON
> > fatal: not a git repository (or any of the parent directories): .git
> > Git Hash: 0000000
> > Experiment folder: DEFINE THIS/ljspeech-ddc-June-23-2021_01+10PM-0000000
> > fatal: not a git repository (or any of the parent directories): .git
> > Traceback (most recent call last):
> > File "/content/TTS/TTS/bin/train_tacotron.py", line 737, in 
> > args, config, OUT_PATH, AUDIO_PATH, c_logger, tb_logger = init_training(sys.argv)
> > File "/content/TTS/TTS/utils/arguments.py", line 182, in init_training
> > config, OUT_PATH, AUDIO_PATH, c_logger, tb_logger = process_args(args)
> > File "/content/TTS/TTS/utils/arguments.py", line 168, in process_args
> > copy_model_files(config, experiment_path, new_fields)
> > File "/content/TTS/TTS/utils/io.py", line 42, in copy_model_files
> > copy_stats_path,
> > File "/usr/lib/python3.7/shutil.py", line 120, in copyfile
> > with open(src, 'rb') as fsrc:
> > FileNotFoundError: [Errno 2] No such file or directory: 'scale_stats.npy'`

eidt your config file and set your` "stats_path": null`

also you need to define dataset path and your output folder in the config file as well.

**Reply by thomasvonl** [reply]
the config file is /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json right? because I have found nothing about `stats_path` there

**Reply by Sadam1195** [reply]
Yes there is, see the line 32.

**Reply by ragireddydpriya** [reply]
Hello @Sadam1195,

Iam also facing similar issue and i've followed up your conversations and came till this point.
After modifying line 32 in /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json, I am facing this issue.

command:

!CUDA_VISIBLE_DEVICES="0" python /content/TTS/TTS/bin/train_tacotron.py --config_path /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json \ --coqpit.output_path ./Results \ --coqpit.datasets.0.path ./content/LJSpeech-1.1/

output of cell:

> Using CUDA:  True
 > Number of GPUs:  1
   >  Mixed precision mode is ON
 > Git Hash: d245b5d4
 > Experiment folder: DEFINE THIS/ljspeech-ddc-July-01-2021_02+44PM-d245b5d4
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:True
 | > symmetric_norm:True
 | > mel_fmin:50.0
 | > mel_fmax:7600.0
 | > spec_gain:1.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:60
 | > do_sound_norm:False
 | > stats_path:None
 | > base:10
 | > hop_length:256
 | > win_length:1024
 ! Run is removed from DEFINE THIS/ljspeech-ddc-July-01-2021_02+44PM-d245b5d4
Traceback (most recent call last):
  File "/content/TTS/TTS/bin/train_tacotron.py", line 739, in <module>
    main(args)
  File "/content/TTS/TTS/bin/train_tacotron.py", line 603, in main
    meta_data_train, meta_data_eval = load_meta_data(config.datasets)
  File "/content/TTS/TTS/tts/datasets/preprocess.py", line 51, in load_meta_data
    meta_data_train = preprocessor(root_path, meta_file_train)
  File "/content/TTS/TTS/tts/datasets/preprocess.py", line 183, in ljspeech
    with open(txt_file, "r", encoding="utf-8") as ttf:
FileNotFoundError: [Errno 2] No such file or directory: 'DEFINE THIS/metadata.csv'


Please do needful to proceed further

Thank you

**Reply by erogol** [reply]
👁️ `FileNotFoundError: [Errno 2] No such file or directory: 'DEFINE THIS/metadata.csv'`

**Reply by ragireddydpriya** [reply]
Hello erogol,

I changed the path of metadata.csv  in /content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json.After modifying fields {audio.stats_path, output_path, phoneme_cache_path, and datasets.path} in .json to specified path , Iam getting below error.

> Using CUDA:  True
 > Number of GPUs:  1
Traceback (most recent call last):
  File "/content/TTS/TTS/bin/train_tacotron.py", line 737, in <module>
    args, config, OUT_PATH, AUDIO_PATH, c_logger, tb_logger = init_training(sys.argv)
  File "/content/TTS/TTS/utils/arguments.py", line 182, in init_training
    config, OUT_PATH, AUDIO_PATH, c_logger, tb_logger = process_args(args)
  File "/content/TTS/TTS/utils/arguments.py", line 145, in process_args
    config = load_config(args.config_path)
  File "/content/TTS/TTS/config/__init__.py", line 62, in load_config
    with open(config_path, "r", encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: './content/TTS/recipes/ljspeech/tacotron2-DDC/tacotron2-DDC.json'

Please kindly suggest ,do i need to modify any fields in config file inorder to train the model.[attaching modified tacotron2-DDC.json
[tacotron2-DDC.zip](https://github.com/coqui-ai/TTS/files/6757911/tacotron2-DDC.zip)
]
